pkg/sql/alter_primary_key.go

```diff
@@ -129,6 +129,12 @@ func (p *planner) AlterPrimaryKey(
 		"new_primary_key",
 		nameExists,
 	)
+	if alterPKNode.Name != "" &&
+		nameExists(string(alterPKNode.Name)) &&
+		// Allow reuse of existing primary key's name.
+		tableDesc.PrimaryIndex.Name != string(alterPKNode.Name) {
+		return errors.Newf("constraint with name %s already exists", alterPKNode.Name)
```

Comment:
is there an appropriate PG error code for this? see `pgcode/codes.go` and `pgerror.Newf()`

Reply:
Thanks for the tip. I used the error code for InvalidName.